File: backend/src/main/java/com/langleon/dsobuildsim/wisdomskilltree/dto/instance/WisdomGroupInstanceDTO.java
package com.langleon.dsobuildsim.wisdomskilltree.dto.instance;

import com.langleon.dsobuildsim.wisdomskilltree.wisdomgroup.WisdomGroupType;
import com.langleon.dsobuildsim.wisdomskilltree.wisdomskill.WisdomSkillType;

import java.util.EnumMap;

public record WisdomGroupInstanceDTO(
        WisdomGroupType type,
        EnumMap<WisdomSkillType, WisdomSkillInstanceDTO> wisdomSkills
) {
}

File: frontend/src/app/models/instanceDTOs/WisdomGroupInstanceDTO.ts
import { WisdomGroupType } from "../../enums/WisdomGroupType";
import { WisdomSkillType } from "../../enums/WisdomSkillType";
import { WisdomSkillInstanceDTO } from "./WisdomSkillInstanceDTO";

export interface WisdomGroupInstanceDTO {
    type: WisdomGroupType;
    wisdomSkills: Partial<Record<WisdomSkillType, WisdomSkillInstanceDTO>>;
}